src/main/java/functional/algebraic/Result.java

```diff
@@ -283,6 +284,27 @@ public void match(Consumer<? super E> ef, Consumer<? super V> vf)
     }
 
     /**
+     * Applies a function that can take any Object to both the error and the result.
+     * Useful for e.g. Systemm.out::println.
+     *
+     * @param f the function to apply
+     */
+    public void collapse(Consumer<Object> f) {
+        either.match(f, f);
+    }
+
+    /**
+     * Returns the value in this Result, or a default if there is an error.
+     *
+     * @param defaultValue the default value
+     * @return the value or default
+     */
+    public V orElse(V defaultValue) {
+        return either.matchThen(ignored -> defaultValue, val -> val);
```

Comment:
Multiline match expressions are preferred for their similarity to haskell `case` syntax; as well as double underscore (`__`) for ignored variables.
```suggestion
        return either.matchThen(
            __ -> defaultValue,
            val -> val
        );
```